### ISSUE TITLE
occasional pipe closed error on concurrent ttc-charts builds

### ISSUE THREAD

**Issue by ryandawsonuk** [Activiti/Activiti#2618]
Error is

```

java.io.IOException: Pipe closed
	at java.io.PipedInputStream.checkStateForReceive(PipedInputStream.java:260)
	at java.io.PipedInputStream.receive(PipedInputStream.java:226)
	at java.io.PipedOutputStream.write(PipedOutputStream.java:149)
	at java.io.OutputStream.write(OutputStream.java:75)
	at org.csanchez.jenkins.plugins.kubernetes.pipeline.ContainerExecDecorator$1.setupEnvironmentVariable(ContainerExecDecorator.java:402)
	at org.csanchez.jenkins.plugins.kubernetes.pipeline.ContainerExecDecorator$1.doLaunch(ContainerExecDecorator.java:348)
	at org.csanchez.jenkins.plugins.kubernetes.pipeline.ContainerExecDecorator$1.launch(ContainerExecDecorator.java:236)
	at hudson.Launcher$ProcStarter.start(Launcher.java:455)
	at org.jenkinsci.plugins.durabletask.BourneShellScript.launchWithCookie(BourneShellScript.java:188)
	at org.jenkinsci.plugins.durabletask.FileMonitoringTask.launch(FileMonitoringTask.java:99)
	at org.jenkinsci.plugins.workflow.steps.durable_task.DurableTaskStep$Execution.start(DurableTaskStep.java:278)
	at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:270)
	at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:178)
	at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:122)
	at sun.reflect.GeneratedMethodAccessor13510.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
```

**Comment by ryandawsonuk** [comment]
Seems to be a known risk/issue with concurrent builds https://github.com/jenkins-x/jx/issues/2370

In this case I think we had multiple builds of the same PR

![image](https://user-images.githubusercontent.com/22754674/54020683-3dd05f80-4186-11e9-99fe-653b98de332a.png)

This is probably because we don't have the convergence checks on TTC

But this wasn't happening before so presumably due to increasing the timeouts as a workaround for https://github.com/Activiti/Activiti/issues/2612

**Comment by ryandawsonuk** [comment]
Now put several concurrent builds through to test:

![image](https://user-images.githubusercontent.com/22754674/54021133-5bea8f80-4187-11e9-83de-0732035c6970.png)

I am not sure why PR-25 ran actually as I didn't trigger that. Might be because I changed the Jenkinsfile in master. I guess closing outstanding PRs would be wise.

Didn't hit the above issue but did find that one of the helm installs failed with `Error: release ttcexample-master-75 failed: timed out waiting for the condition
make: *** [install] Error 1`

Best guess as to why is that a particular node was maxed out:

![image](https://user-images.githubusercontent.com/22754674/54021302-b8e64580-4187-11e9-90eb-1228fd25d6b9.png)

I guess this means k8s here is not making full use of the resources available and distributing ineffectively. That would have to be due to the way we've set the requests and limits.

The numbers on `kubectl describe nodes` don't seem particularly high except for the maven build pods so will set those in line with https://github.com/Activiti/Activiti/issues/2582#issuecomment-469198535
